gRPC/src/main/kotlin/rys/ajaxpetproject/grpc/MessageGrpc.kt

```diff
@@ -0,0 +1,147 @@
+package rys.ajaxpetproject.grpc
+
+import io.nats.client.Connection
+import net.devh.boot.grpc.server.service.GrpcService
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toFlux
+import reactor.kotlin.core.publisher.toMono
+import reactor.util.context.Context
+import rys.ajaxpetproject.commonmodels.message.proto.MessageDto
+import rys.ajaxpetproject.request.message.create.proto.MessageCreateResponse
+import rys.ajaxpetproject.request.message.create.proto.MessageCreateRequest
+import rys.ajaxpetproject.service.ChatService
+import rys.ajaxpetproject.service.MessageService
+import rys.ajaxpetproject.service.message.ReactorMessageServiceGrpc
+import rys.ajaxpetproject.utils.toModel
+import rys.ajaxpetproject.internalapi.MessageEvent
+import rys.ajaxpetproject.model.MongoMessage
+import rys.ajaxpetproject.request.message.subscription.proto.EventSubscription.CreateSubscriptionRequest
+import rys.ajaxpetproject.request.message.subscription.proto.EventSubscription.CreateSubscriptionResponse
+import rys.ajaxpetproject.utils.toProto
+
+@GrpcService
+@Suppress("TooGenericExceptionCaught")
+class MessageGrpc(
+    natsConnection: Connection,
+    private val chatService: ChatService,
+    private val messageService: MessageService,
+) : ReactorMessageServiceGrpc.MessageServiceImplBase() {
+
+    private val messageParser = MessageDto.parser()
+    private val dispatcher = natsConnection.createDispatcher()
```

Comment:
you're mixing grpc and nats here, consider wrap interaction with nats in some class